DifferentialDrive.md

```diff
@@ -0,0 +1,545 @@
+- `gyroSim.setAngle`: Updates the simulated gyro angle based on the robot’s simulated heading.
+
+This method ensures that our simulated sensors provide accurate feedback as the robot "moves" in the simulation, allowing us to test and tweak our code.
+
+#### Logging: Capturing Important Data
+Now let’s talk about logging. Logging is crucial for understanding how our robot behaves over time, diagnosing issues, and improving performance. We’re going to use Monologue for logging, which gives us a structured way to record and analyze data from the robot’s systems.
+
+Here’s how we set up Monologue:
+```java
+private void configureGameBehavior() {
+  Monologue.setupMonologue(this, "/Robot", false, true);
+  addPeriodic(Monologue::updateAll, kDefaultPeriod);
+  addPeriodic(FaultLogger::update, 1);
+}
+```
+This initializes Monologue with our robot, setting up the logging system to capture data as the robot runs and makes sure Monologue logs data at regular intervals, which we’ve defined with kDefaultPeriod. By logging data regularly, we can later review how the robot performed and make informed adjustments to our code.
+
+#### Using @Log.NT for NetworkTables Logging
+We use NetworkTables to make logging specific variables or objects easier. We’re going to use the `@Log.NT` annotation as it allows us to log key metrics in real-time while the robot is running, which we can then view on dashboards like Shuffleboard or SmartDashboard.
+
+Lets get started by making a field:
+```java
+@Log.NT private final Field2d field2d = new Field2d();
+```
+Read up on the Sim. reference sheet to see when and what we should be logging through network tables. 
+
+One other thing we're going to log is going to be our position on the field. Simply use the annotation on the `getPose` method:
+```java
+@Log.NT
+public Pose2d getPose() {
+  return odometry.getPoseMeters();
+}
+```
+Another quick little thing we're going to log is to see if our motors are reaching their setpoints. This is done by returning the `atSetpoint` from both of the pid controllers and log the data:
+```java
+@Log.NT
+public boolean atSetpoint() {
+  return leftPIDController.atSetpoint() && rightPIDController.atSetpoint();
+}
+```
+By logging these values, we can gain real-time insights into our robot’s performance, which is essential for making data-driven decisions and quickly diagnosing issues.
+
+#### Updating the Robot: periodic
+Finally, the periodic method is where we update various elements of the robot regularly during operation. It’s like the heartbeat of our subsystems. Specifically, we are going to update the field display with the robot’s current position, which helps us visualize its movement.
+```java
+field2d.setRobotPose(getPose());
+```
+
+#### Seeing the result
+Start by launching sim and opening up NetworkTables. In which you will see a Robot field, which will contain all of the things we've logged.
```

Comment:
Since it's their first time opening up sim, they probably won't immediately see the field widget displayed. Link them to the sim gui doc and guide them on where to find it (smartdashboard iirc).

Reply:
Not sure if those local images might show that, but they're not on the repo so they won't show until uploaded.